From eb1ffa36c8cc237102a70294efb6be88917a577b Mon Sep 17 00:00:00 2001
From: "haotian.qu" <haotian.qu@amlogic.com>
Date: Wed, 26 Apr 2023 10:55:32 +0800
Subject: [PATCH] SWUpdate: upgrade fails by network [1/1]

PD#SWPL-121506

Problem:
Upgrade fails by network.

Solution:
Kernel 5.4 uses /proc/inand nodes to determine
emmc or nand, which no longer works in kernel 5.15

Verify:
t7c local

Signed-off-by: haotian.qu <haotian.qu@amlogic.com>
Change-Id: I7d09172adc68f74b8a8f24442578a49d97f45ac5
---
 .../web-ui-wifi/files/cgi-bin/scripts/swupdate/swupdate.sh      | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/recipes-connectivity/web-ui-wifi/files/cgi-bin/scripts/swupdate/swupdate.sh b/recipes-connectivity/web-ui-wifi/files/cgi-bin/scripts/swupdate/swupdate.sh
index 0f8a960..6b4ba27 100755
--- a/recipes-connectivity/web-ui-wifi/files/cgi-bin/scripts/swupdate/swupdate.sh
+++ b/recipes-connectivity/web-ui-wifi/files/cgi-bin/scripts/swupdate/swupdate.sh
@@ -2,7 +2,7 @@
 PATH=/bin:/sbin:/usr/bin:/usr/sbin
 
 if [ $1 == "start" ];then
-  if [ -f "/proc/inand" ]; then
+  if [ -d "/sys/block/mmcblk0" ]; then
     swupdate --recovery -l 6 -k /etc/swupdate-public.pem -w "-r /var/www/swupdate/" > /tmp/swupdate.log &
   else
     swupdate --recovery -l 6 -k /etc/swupdate-public.pem -b "0 1 2 3 4" -w "-r /var/www/swupdate/" > /tmp/swupdate.log &
-- 
2.25.1

